src/smk/tool/geomark/tool-geomark.js

```diff
@@ -211,14 +211,45 @@ include.module( 'tool-geomark', [
             }
 
             this.toGeomark = function(geomarkInfo, drawingLayer) {
+                const geomarkProperties = this.getGeomarkProperties(geomarkInfo);
                 return {
-                    id: geomarkInfo.id || geomarkInfo.properties.id,
-                    url: geomarkInfo.url || geomarkInfo.properties.url,
+                    id: geomarkInfo.id || geomarkProperties.id,
+                    url: geomarkInfo.url || geomarkProperties.url,
                     drawingLayer: drawingLayer,
                     isVisible: true
                 };
             }
 
+            this.getGeomarkProperties = function(geomarkInfo) {
+                // polygon
+                if (geomarkInfo.properties) {
+                    return geomarkInfo.properties;
+                }
+                // multi-polygon
+                if (geomarkInfo.features && geomarkInfo.features[0] && geomarkInfo.features[0].properties) {
```

Comment:
I'm having a forgetful moment, will indexing into an empty array like this throw an exception in JavaScript? I might just be thinking about other languages (like C#).